site/src/app/room/[room_id]/page.tsx

```diff
@@ -5,31 +5,80 @@ import { Footer } from '@/components/Footer';
 import { Buttons } from '@/components/Buttons';
 import { CopyLink } from '@/components/CopyLink';
 import { Display } from '@/components/Display';
-import { getBaseUrl } from '@/lib/server/getBaseUrl';
-const nextQuestionHandler = () => {
-	console.log('test');
-};
+import { usePathname } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import { pusherClient } from '@/lib/pusher';
+
 
 export default function Home() {
-	return (
-		<>
-			<main className='flex min-h-screen flex-col items-center justify-between p-24'>
-				<Header />
-				<Display text='LINK / QUESTION' />
-				<CopyLink hostName={getBaseUrl()} />
-				<Buttons text='Next Question' size='lg' onClick={nextQuestionHandler} />
-				<Footer />
-
-				{/* Will be replace by link or display component  */}
-				{/* <div className="relative flex place-items-center before:absolute before:h-[300px] before:w-[480px] before:-translate-x-1/2 before:rounded-full before:bg-gradient-radial before:from-white before:to-transparent before:blur-2xl before:content-[''] after:absolute after:-z-20 after:h-[180px] after:w-[240px] after:translate-x-1/3 after:bg-gradient-conic after:from-sky-200 after:via-blue-200 after:blur-2xl after:content-[''] before:dark:bg-gradient-to-br before:dark:from-transparent before:dark:to-blue-700 before:dark:opacity-10 after:dark:from-sky-900 after:dark:via-[#0141ff] after:dark:opacity-40 before:lg:h-[360px] z-[-1]">
-					LINK / QUESTION
-				</div> */}
-				{/*   */}
-				{/* Will be replace by button component  */}
-				{/* <div className='relative flex place-items-center before:absolute '>
-					<Buttons text='next question' size='lg' onClick={() => console.log('Next question')} />
-				</div> */}
-			</main>
-		</>
-	);
+
+	const path = usePathname();
+	const roomId = (path.split('/')[2])
+
+	const [questionText, setQuestionText] = useState('')
+	const [currentQuestionNumber, setCurrentQuestionNumber] = useState(0)
+
+
+	useEffect(() => {
+		// getting the current question text of the room
+		fetch(`/api/room/${roomId}`)
+		.then(response => response.json())
+		.then(data => {
+			setQuestionText(data.current_question_text);
+		})
+		.catch(error => console.error('Error:', error));
+
+		// subcribe to the proper channel
+		pusherClient.subscribe(`${roomId}`)
+
+		// question text coming from the server
+		const questionTextHandler = (question_text: string) => {
+			setQuestionText(() => question_text)
+		}
+
+		pusherClient.bind('next-question', questionTextHandler)
+
+		return () => {
+			pusherClient.unsubscribe(`${roomId}`)
+			pusherClient.unbind('next-question', questionTextHandler)
+		}
+	},[])
+
+
+	const nextQuestionText = async () => {
+		await fetch(`/api/room/${roomId}`, {
+			method: 'PATCH',
+		  })
+		  .then((response) => {
```

Comment:
No need to do anything with the response since the server sends the questoin through the pusher channel

Reply:
Thanks for the review, Sabih! I made some updates according to your feedback and just pushed those changes up. Please let me know if you have any other feedback!